### PR TITLE
dotnetCorePackages.dotnet_{8,9}.vmr: switch CDN URL

### PR DESCRIPTION
**Pull request by GGG-KILLER**
As it was announced in dotnet/announcements#336, the CDN for .NET assets is changing, so change those ahead of time.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by JamieMagee — APPROVED** [review]

**Comment by JamieMagee** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367825`

File: pkgs/development/compilers/dotnet/update.nix [hunk 1]
```diff
@@ -129,6 +129,7 @@ writeScript "update-dotnet-vmr.sh" ''
               map(select(.PrivateSourceBuiltArtifactsUrl))
               | .[] | .PrivateSourceBuiltArtifactsUrl' eng/Versions.props)
       fi
+      artifactsUrl="''${artifactsUrl/dotnetcli.azureedge.net/builds.dotnet.microsoft.com}"
```
Comment (corngood) [inline pkgs/development/compilers/dotnet/update.nix:132]: we should maybe leave a comment indicating this is temporary until the release tarballs are fixed?

Comment (GGG-KILLER) [reply pkgs/development/compilers/dotnet/update.nix:132]: Given they won't use these URLs ever again, I don't know if it is really temporary.

They said they don't know for how long they'll be able to use the `*.azureedge.net` domain for.

**Comment by GGG-KILLER** [comment]
> can we fix the:
> 
> ```
> artifactsUrl=https://dotnetcli.azureedge.net/source-built-artifacts/assets/Private.SourceBuilt.Artifacts.$artifactsVersion.centos.9-x64.tar.gz
> ```
> 
> as well so it works if we later remove the substitution below?

done

File: pkgs/development/compilers/dotnet/update.nix (hunk 1 above)
Comment (corngood) [reply pkgs/development/compilers/dotnet/update.nix:132]: I just mean presumably they will update the urls in the VMR repo in the future, so we won't need to replace them.

Comment (GGG-KILLER) [reply pkgs/development/compilers/dotnet/update.nix:132]: True, I was thinking more of previous versions but we don't need to keep supporting those now that I think of it.

**Pull request merged**